README.md

```diff
@@ -0,0 +1,341 @@
+   working with separate APIs.
+
+
+### Why GitHub?
+
+We chose to use Git and GitHub as the primary tool for working and hosting
+*documentation for developers*:
+
+ * All developers are familiar with Git. Many developers are also familiar with
+   GitHub. **We will require developers to review and accept documentation
+   often** throughout the project. We believe that these tools will help them
+   with this task.
+
+ * Git allows us to easily keep track of all the changes. Having the option
+   to produce a set of changes between particular versions of the documentation
+   (e.g. "what has changed between versions `1.0.0` and `1.2.7`?") is a very
+   important feature, and it can be easily achieved with Git.
+
+ * GitHub features include an integrated code review tools, issue trackers,
+   change notifications, automated build tools - and many other features. We
+   are aware that most of our repositories will be hosting documentation
+   only, but we do find all these tools useful even in documentation-only
+   projects.
+
+
+Issue tracker
+-------------
+
+GitHub offers built-in issue tracking which should be entirely sufficient for
+purposes of developing the software and its documentation.
+
+Every GitHub project may have its own separate issue tracker, and we have
+chosen to stick to this design decision. In other words, we will use **multiple
+issue trackers**. When you're reading documentation on GitHub and you find a
+bug, simply scroll to the top of the page, and click the *Issues* tab.
+
+Pros:
+
+ * We keep project's issues along with the project's code.
+ * The *Issues* tab is always around.
+
+Cons:
+
+ * Issue numbers will not be unique within the entire EWP Project. When
+   cross-referencing issues between sub-projects, absolute issue URLs will need
+   to be used. However, since WP6 will also use a separate issue tracker (and
+   we were [unable to share it]
+   (https://github.com/erasmus-without-paper/general-issues/issues/1)), the
+   issue numbers would not be unique *in either case*.
+
+
+Rules for API Design and Versioning
+-----------------------------------
+
+### Semantic Versioning
+
+We will use **Semantic Versioning** for releases of EWP technical
+documentation. It has enormous advantages, when it comes to working (and
+reworking) API design:
+
+ * We will assign documents their version numbers only after they are
+   officially approved.
+
+ * Semantic versioning scheme allows developers to easily determine when a
+   breaking change occurs in a particular API.
+
+ * It follows [a set of quite explicit rules](http://semver.org/). Please get
+   familiar with them (especially if you plan to design parts of the APIs).
+
+ * Having a single, official latest version of the document, combined with the
+   requirement of preparing an explicit and strict API specifications, allows us
+   to easily answer the question: **if it doesn't work then who's fault is
+   that?**
+
```

Comment:
On 05/02/16 08:46, Wojciech Rygielski wrote:
> In README.md

>> + * Having a single, official latest version of the document, combined with the
>> +   requirement of preparing an explicit and strict API specifications, allows us
>> +   to easily answer the question: **if it doesn't work then who's fault is
>> +   that?**
>> +
> 
> I know, it did sound a little "unpleasant" when I was typing that down.
> I'll try to fix this. The "who" in this sentence was not really supposed
> to be a specific person. But for each request, we need to know if it is
> the clients who needs fixing or the server. If we cannot determine that,
> then the specification is flawed, and - sooner or later - it will need
> to be fixed. Do you agree?

I do agree. I think that clarifying that it is the "entity" (client or
server) and not the programmer you are referring to, is the solution to
this.

-- 
Victoriano Giralt                             Central ICT Services
Systems Manager                               University of Malaga
+34952131415                                  SPAIN
==================================================================
Note: signature.asc is the electronic signature of present message
A: Yes.
> Q: Are you sure ?
>> A: Because it reverses the logical flow of conversation.
>>> Q: Why is top posting annoying in email ?